web/client/codechecker_client/blame_info.py

```diff
@@ -120,6 +120,13 @@ def assemble_blame_info(
     # are lost.
     if sys.platform == "win32":
         file_blame_info = __collect_blame_info_for_files(file_paths)
+    elif sys.platform == "darwin":
+        # multiprocess might not be available on windows
+        # we import it here
+        import multiprocess
+        with multiprocess.Pool() as pool:
+            file_blame_info = __collect_blame_info_for_files(
+                file_paths, pool.map)
     else:
         with ProcessPoolExecutor() as executor:
```

Comment:
But why duplicate this logic?

At the top of the file, after all other imports:

```py
if sys.platform == "darwin":
  from multiprocessing import Pool as MultiprocessingPool
else:
  from concurrent.futures import ProcessPoolExecutor as MultiprocessingPool
```

If we want to be extra pedantic:

```py
if sys.platform == "win32":
  MultiprocessingPool = None
elif sys.platform == "darwin":
  from multiprocessing import Pool as MultiprocessingPool
else:
  from concurrent.futures import ProcessPoolExecutor as MultiprocessingPool
```
and then here simply this, because you no longer need to handle "darwin" and <anything that isn't "win32" and "darwin"> separately:

```suggestion
    else:
        with MultiprocessingPool() as pool:
            file_blame_info = __collect_blame_info_for_files(
                file_paths, pool.map)
```

Reply:
Actually, you could get even more pedantic (I see this is used in two files) by creating a compatibility module which imports the right thing based on platform, and for `win32`, defines itself as a context manager which does nothing when entering and exiting. (Not sure how to deal with the `.map` member...)

That would let us get rid of all the platform-specific distinctions in all the now affected locations. Compatibility shims are not the devil, and given that all "import" does is "set this module-global variable to point at a special dictionary object", you can make everything more readable by getting rid of conditions and indirections.

Reply:
I had the false impression, that the multiprocess lib is not supported on Windows, that's why the conditional import.
In the latest version, I completely eliminated the branch around the logic, and use multiprocess.Pool on all platforms.